hugo/assets/scripts/statemachine.tsx

```diff
@@ -0,0 +1,356 @@
+    super(props);
+    this.state = {
+      name: props.name,
+      isEnabled: props.isEnabled,
+    };
+  }
+
+  /**
+   * Enables or disables the event button
+   * @param enabled true if the button should be enabled
+   */
+  setEnabled(enabled: boolean) {
+    this.setState({ isEnabled: enabled });
+  }
+
+  render() {
+    return (
+      <button
+        onClick={this.props.handleClick}
+        disabled={this.state.isEnabled}
+        className="text-white border-2 border-solid bg-emeraldLangiumABitDarker rounded-md p-4 text-center text-sm enabled:hover:shadow-opacity-50 enabled:hover:shadow-[0px_0px_15px_0px] enabled:hover:shadow-emeraldLangium disabled:border-gray-400 disabled:text-gray-400 disabled:bg-emeraldLangiumDarker ">
+        {this.props.name}
+      </button>
+    );
+  }
+}
+
+class Preview extends React.Component<PreviewProps, PreviewProps> {
+  constructor(props: PreviewProps) {
+    super(props);
+    this.state = {
+      diagnostics: props.diagnostics,
+      astNode: props.astNode,
+    };
+
+    this.startPreview = this.startPreview.bind(this);
+  }
+
+  startPreview(ast: StateMachineAstNode, diagnostics: Diagnostic[]) {
+    this.setState({astNode: ast, diagnostics: diagnostics });
+  }
+
+  render() {
+    // check if code contains an astNode
+    if (!this.state.astNode) {
+      // Show the exception
+      return (
+        <div className="flex flex-col h-full w-full p-4 justify-start items-center my-10">
+          <div className="text-white border-2 border-solid border-accentRed rounded-md p-4 text-center text-sm cursor-default">
+            No Ast found
+          </div>
+        </div>
+      );
+    }
+
+    // if the code doesn't contain any errors
+    if (this.state.diagnostics == null || (this.state.diagnostics.length == 0)) {
+      let states: State[] = [];
+      let events: Event[] = [];
+
+      let statemachineTools = new StateMachineTools(this.state.astNode);
+      // update the aktive state
+      const changeStates = function(state: StateMachineState) {
+        statemachineTools.setState(state);
+
+        // loop through all states and set the active state
+        events.forEach((i) => {
+          i.setEnabled(statemachineTools.isEventEnabled(statemachineTools.getEventByName(i.props.name)!));
+        });
+
+        states.forEach((i) => {
+          i.setActive(statemachineTools.isCurrentState(statemachineTools.getStateByName(i.props.name)!));
+        });
+   
+      }
+
+      return (
+        <div className="flex flex-col h-full w-full p-4 float-right items-center">
+          <p className="text-white text-lg w-full my-4">Events</p>
+          <div className="flex flex-wrap w-full gap-2">
+            {statemachineTools.getEvents().map((event, index) => {
+              return (
+                <Event
+                  isEnabled={statemachineTools.isEventEnabled(event)}
+                  handleClick={() => changeStates(statemachineTools.getNextState(event)!)}
```

Comment:
I would be a bit careful with using non-null assertion op so frequently (`!`), as it can hide some issues down the road. Doesn't necessarily need to be addressed here however.